### PR TITLE
Pullupstream/v0.5.0

### PR DESCRIPTION
**Pull request by rachit77**
This PR fetches v0.5.0 changes of the ZK-EVM node and bumps the [cdk-data-availiability](https://github.com/0xPolygon/cdk-data-availability) dependency to v0.4.0.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+name: release
```
Comment (arnaubennassar) [inline .github/workflows/release.yml]: remove this file, it's used for Hermez releases, we should have our own thing

File: cmd/main.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ var (
 	networkFlag = cli.StringFlag{
 		Name:     config.FlagNetwork,
 		Aliases:  []string{"net"},
-		Usage:    "Load default network configuration. Supported values: [`mainnet`, `testnet`, `custom`]",
+		Usage:    "Load default network configuration. Supported values: [`mainnet`, `testnet`, `cardona`, `custom`]",
```
Comment (arnaubennassar) [inline cmd/main.go]: Didn't realise about this before, but we should only support `custom` option

File: config/cardonagenesis.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
+package config
```
Comment (arnaubennassar) [inline config/cardonagenesis.go]: remove this file

File: config/config.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ const (
 	FlagYes = "yes"
 	// FlagCfg is the flag for cfg.
 	FlagCfg = "cfg"
-	// FlagNetwork is the flag for the network name. Valid values: ["testnet", "mainnet", "custom"].
+	// FlagNetwork is the flag for the network name. Valid values: ["testnet", "mainnet", "cardona", "custom"].
```
Comment (arnaubennassar) [inline config/config.go]: in the coment, only custom

File: config/environments/cardona/example.env [hunk 5]
```diff
@@ -0,0 +1,9 @@
+ZKEVM_NETWORK = "cardona"
```
Comment (arnaubennassar) [inline config/environments/cardona/example.env]: remove:
- config/environments/cardona
- config/environments/testnet
- config/environments/mainnet

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

File: config/environments/testnet/postgresql.conf [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# -----------------------------
+-----------------------------
```
Comment (christophercampbell) [inline config/environments/testnet/postgresql.conf]: This should be comment line

**Review by christophercampbell — DISMISSED** [review]
LGTM other than missing comment prefix in config file. 

And one comment: I presume the the migration changes are coming from upstream? At some point, once we have installs, we should consider migration files fixed and only accept/write new migrations files.

**Comment by christophercampbell** [comment]
If the testnet configuration is breaking CI, it might be because the postgres conf file is borked (see above)

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by begmaroman — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Comment by Stefan-Ethernal** [comment]
> If the testnet configuration is breaking CI, it might be because the postgres conf file is borked (see above)

Nope, this is not the case, we have removed all testnet configuration and ZK-EVM node e2e tests are green, only the one which tests DAC is failing (since DA nodes are failing to start up with the error suggesting that contract is not deployed on an expected address):

```bash
2024-02-05T16:41:14.975Z	[31mFATAL[0m	cmd/main.go:104	no contract code at given address
/src/log/log.go:140 github.com/0xPolygon/cdk-data-availability/log.appendStackTraceMaybeArgs()
/src/log/log.go:221 github.com/0xPolygon/cdk-data-availability/log.Fatal()
/src/cmd/main.go:104 main.start()
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/command.go:274 github.com/urfave/cli/v2.(*Command).Run()
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/command.go:267 github.com/urfave/cli/v2.(*Command).Run()
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/app.go:332 github.com/urfave/cli/v2.(*App).RunContext()
/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/app.go:309 github.com/urfave/cli/v2.(*App).Run()
/src/cmd/main.go:52 main.main()
/usr/local/go/src/runtime/proc.go:250 runtime.main()
	{"pid": 1, "version": "v0.0.3-30-gc030128"}
main.start
```

**Pull request merged**